requirements.txt

```diff
@@ -6,6 +6,9 @@ cirq-unstable==0.9.0.dev20200813185757
 pyquil~=2.18.0
 qiskit~=0.16.2
 
+# to handle deprecations
+deprecation~=2.1.0
```

Comment:
this is very cool, but it will need to be in `install_requires` of the `setup.py` (otherwise `mitiq` will break), and I'm not sure that it really makes life so much easier that it's worth the additional dependency

Reply:
What's the alternative for handling deprecation that doesn't require an additional dependency?

Reply:
Typically one would just use the `warnings` library to raise a `DeprecationWarning` like we do in pyQuil [here](https://github.com/rigetti/pyquil/blob/6f5c9b1a3aced2b5cec5d81859d81c0e339d16fb/pyquil/device/_main.py#L106). The `deprecations` decorator likely does exactly this, but with a nicely formatted string with the version numbers indicating when to expect the feature to be removed (which you can do yourself with the string argument to `warn`). Lmk if you have any more qs!